experimental/QuadFormAndIsom/src/enumeration.jl

```diff
@@ -0,0 +1,952 @@
+  s = length(inf)
+  signs = Dict{typeof(inf[1]), Int}[]
+  parts = Vector{Int}[]
+  perm = AllPerms(s)
+  for v in AllParts(l)
+    if any(i -> i > rk, v)
+      continue
+    end
+    if length(v) > s
+      continue
+    end
+    while length(v) != s
+      push!(v, 0)
+    end
+    for vv in perm
+      v2 = v[vv.d]
+      v2 in parts ? continue : push!(parts, v2)
+    end
+  end
+  for v in parts
+    push!(signs, Dict(a => b for (a,b) in zip(inf, v)))
+  end
+  return signs
+end
+
+@doc raw"""
+    representatives_of_hermitian_type(Lf::ZZLatWithIsom, m::Int = 1)
+                                            -> Vector{ZZLatWithIsom}
+
+Given a lattice with isometry $(L, f)$ of hermitian type (i.e. the minimal
+polynomial of `f` is irreducible cyclotomic) and a positive integer `m`, return
+a set of representatives of isomorphism classes of lattices with isometry of
+hermitian type $(M, g)$ and such that the type of $(B, g^m)$ is equal to the
+type of $(L, f)$. Note that in this case, the isometries `g`'s are of
+order $nm$.
+
+# Examples
+```jldoctest
+julia> L = root_lattice(:A,2);
+
+julia> Lf = integer_lattice_with_isometry(L);
+
+julia> reps = representatives_of_hermitian_type(Lf, 6)
+1-element Vector{ZZLatWithIsom}:
+ Integer lattice with isometry of finite order 6
+
+julia> is_of_hermitian_type(reps[1])
+true
+```
+"""
+function representatives_of_hermitian_type(Lf::ZZLatWithIsom, m::Int = 1)
+  rank(Lf) == 0 && return ZZLatWithIsom[Lf]
+
+  @req m >= 1 "m must be a positive integer"
+  @req is_of_hermitian_type(Lf) "Lf must be of hermitian type"
+
+  rk = rank(Lf)
+  d = det(Lf)
+  n = order_of_isometry(Lf)
+  s1, _, s2 = signature_tuple(Lf)
+
+  reps = ZZLatWithIsom[]
+
+  if n*m < 3
+    @vprint :ZZLatWithIsom 1 "Order smaller than 3\n"
+    f = (-1)^(n*m+1)*identity_matrix(QQ, rk)
+    G = genus(Lf)
+    repre = representatives(G)
+    @vprint :ZZLatWithIsom 1 "$(length(repre)) representative(s)\n"
+    for LL in repre
+      is_of_same_type(Lf, integer_lattice_with_isometry(LL, f^m, check=false)) && push!(reps, integer_lattice_with_isometry(LL, f, check=false))
+    end
+    return reps
+  end
+
+  !iseven(s2) && return reps
+
+  @vprint :ZZLatWithIsom 1 "Order bigger than 3\n"
+  ok, rk = Hecke.divides(rk, euler_phi(n*m))
+  ok || return reps
+
+  gene = Hecke.HermGenus[]
+  E, b = cyclotomic_field_as_cm_extension(n*m)
+  Eabs, EabstoE = absolute_simple_field(E)
+  DE = EabstoE(different(maximal_order(Eabs)))
+
+  @vprint :ZZLatWithIsom 1 "We have the different\n"
+
+  ndE = d*inv(QQ(absolute_norm(DE)))^rk
+  detE = _ideals_of_norm(E, ndE)
+
+  @vprint :ZZLatWithIsom 1 "All possible ideal dets: $(length(detE))\n"
+
+  signatures = _possible_signatures(s1, s2, E, rk)
+
+  @vprint :ZZLatWithIsom 1 "All possible signatures: $(length(signatures))\n"
+  for dd in detE, sign in signatures
+    append!(gene, hermitian_genera(E, rk, sign, dd, min_scale = inv(DE), max_scale = numerator(dd)*DE))
+  end
+  gene = unique(gene)
```

Comment:
```suggestion
  unique!(gene)
```
this would be more efficient, if you do not need a copy